### PR TITLE
Guard dedup candidate author scoring against inflation

### LINKED ISSUE #544: Guard dedup candidate author scoring (dis_max, Unicode tokenization) + config docs
## Summary

Author matching in dedup candidate search currently inflates scores for large author lists and drops non‑ASCII names. This work introduces dis_max author matching with bounded scoring, Unicode-aware tokenization, and configurable limits. It also updates env examples/docs to expose the new settings.

## Motivation

- `bool.should` over many authors causes score inflation (large collaborations dominate).
- Current tokenization drops non‑ASCII names.
- Limits/filters should be configurable and documented.

## Scope

- Build author clauses via `dis_max` instead of `bool.should`.
- Cap author clauses and filter short tokens to reduce noise.
- Support Unicode tokens, including CJK.
- Deduplicate identical author token strings.
- Add configuration for author clause cap and min token length.
- Document `DEDUP_SCORING` in `.env.example` and README.

## Proposed Implementation

- **`app/domain/references/repository.py`**
    - Add `_build_author_dis_max_query` with tokenization, dedupe, cap, and dis_max.
    - Replace author `should` clause with dis_max result.
    - Use `_TOKEN_PATTERN = re.compile(r"[^\W\d_]+", flags=re.UNICODE)` and `_is_meaningful_token` to preserve non‑Latin single‑char tokens and exclude Latin initials.
- **`app/core/config.py`**
    - Add `DedupScoringConfig` and `Settings.dedup_scoring`.
- **`tests/unit/domain/references/deduplication/test_candidate_search.py`**
    - Coverage for empty/large author lists, cap behavior, token filtering, Unicode names, and accented initials.
- **`.env.example` + `README.md`**
    - Add `DEDUP_SCORING` example.

## Acceptance Criteria

1. Author scoring no longer inflates for long author lists (dis_max used).
2. Author clause count is capped and configurable.
3. Non‑ASCII author names are tokenized and matched.
4. Tests cover new logic and edge cases (initials, cap behavior, Unicode).
5. `DEDUP_SCORING` is documented in `.env.example` and README.

## Notes / Risks

- Behavior change: no global `minimum_should_match` across authors. If needed, add per‑author match strictness (`operator="and"` or `minimum_should_match` on each match).

## Tests

```bash
pytest tests/unit/domain/references/deduplication/test_candidate_search.py
```

### PR DESCRIPTION
**Pull request by jaybea**
## Summary

- Replace `bool.should` with `dis_max` for author matching in dedup candidate search, preventing score inflation from large author lists (e.g. 2,927-author CERN papers)
- Add Unicode-aware tokenization to handle non-Latin author names (accented, CJK)
- Make author clause cap and min token length configurable via `DedupScoringConfig`

Closes #544

## Changes

- **`app/domain/references/repository.py`** — `_build_author_dis_max_query` helper with dis_max, token filtering, dedup, and clause cap. Replaces N `bool.should` author clauses and removes `minimum_should_match`.
- **`app/core/config.py`** — `DedupScoringConfig` with `max_author_clauses` (default 25) and `min_author_token_length` (default 2).
- **`tests/unit/domain/references/deduplication/test_candidate_search.py`** — 12 tests covering clause cap, token filtering, Unicode names, CJK preservation, and edge cases.
- **`.env.example`** / **`README.md`** — Document `DEDUP_SCORING` config.

## Test plan

- [x] `pytest tests/unit/domain/references/deduplication/test_candidate_search.py` — 12 passed
- [x] `pytest` full suite — 474 passed, 3 pre-existing auth mock failures on main
- [x] `pre-commit run --all-files` clean

### CODE REVIEW

File: app/domain/references/repository.py [hunk 1]
```diff
@@ -231,6 +233,77 @@ async def find_with_identifiers(
...
+
+    if len(token) != 1:
+        return False
+
+    # Preserve single-character non-Latin tokens (e.g., CJK names),
+    # but exclude Latin initials even when accented.
+    return "LATIN" not in unicodedata.name(token, "")
+
+
+def _build_author_dis_max_query(
```
Comment (Adam-Hammo) [inline app/domain/references/repository.py]: Nitpick - would this best live as a classmethod inside the `ReferenceESRepository`?
